spec/models/subject_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'rails_helper'
+
+RSpec.describe Subject, type: :model do
+  describe '#approved_credits' do
+    let(:s1) { create :subject, credits: 10 }
+    let(:s2) { create :subject, credits: 11 }
+    let(:s3) { create :subject, :with_exam, credits: 12 }
+
+    it 'returns approved credits' do
+      expect(Subject.approved_credits([])).to eq(0)
```

Comment:
the `let`s above will be created lazily, so when this expect executes, none of the subjects will be created.
I wonder if we should use `let!` so all subjects are created beforehand